src/xpk/parser/cluster.py

```diff
@@ -904,6 +905,11 @@ def add_shared_cluster_create_capacity_arguments(
           ' types.'
       ),
   )
+  parser_or_group.add_argument(
+      '--managed-ml-diagnostics',
+      action='store_true',
+      help='Enables the installation of required ML Diagnostics components.',
+  )
```

Comment:
You added this in a wrong place. Please put the flag inside `add_shared_cluster_create_optional_arguments ` not `add_shared_cluster_create_capacity_arguments`.